### PR TITLE
Fix hasItemStack in racks

### PR DESCRIPTION
**Pull request by Mekle001**
# Changes proposed in this pull request:
- Fix TileEntityRack::hasItemStack to compare not just key, but quanity
- Without this, the request system has all sorts of weird behavior, and stuck requests

Review please

### CODE REVIEW

File: src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java [hunk 1]
```diff
@@ -133,7 +133,8 @@ public int getFreeSlots()
     @Override
     public boolean hasItemStack(final ItemStack stack, final boolean ignoreDamageValue)
     {
-        return content.get(new ItemStorage(stack, ignoreDamageValue)) != null;
+        final ItemStorage checkItem = new ItemStorage(stack, ignoreDamageValue);
+        return content.containsKey(checkItem) && content.get(checkItem) >= stack.getCount();
```
Comment (someaddons) [inline src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java:137]: don't you need content.get(checkItem).getAmount() ?

Comment (marchermans) [reply src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java:137]: This behaviour is also completely wrong.
The ItemStackRequest, and StackListRequest have flags for checking against tags, metadata and other values optionally.
This is completely ignored in this situation.

Please fix this.

**Review by marchermans — CHANGES_REQUESTED** [review]

File: src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java (hunk 1 above)
Comment (Mekle001) [reply src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java:137]: @someaddons content.get(checkItem) returns an integer which is quantity. So, no, you don't need getAmount. 

@OrionDevelopment this is core TileEntityRack behavior that I'm just restoring previous behavior to. The flags are applied at another layer, and have never been driven down to here.

Comment (marchermans) [reply src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java:137]: Okey.

Comment (marchermans) [reply src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java:137]: Okey.

**Comment by Raycoms** [comment]
Pushed a htofix that reverts the change completely.

**Pull request closed without merging**